### PR TITLE
[Expressions] Add missing tag (crs ellipsoid) to layer_property function

### PR DESCRIPTION
**Pull request by gacarrillor**
Followup #55378

### CODE REVIEW

File: resources/function_help/json/layer_property [hunk 1]
```diff
@@ -20,5 +20,5 @@
     "expression": "layer_property('landsat','crs')",
     "returns": "'EPSG:4326'"
   }],
-  "tags": ["property", "matching", "metadata", "layer name", "layer id", "title", "abstract", "keywords", "data url", "attribution", "attribution url", "layer source", "minimum scale", "min scale", "maximum scale", "max scale", "is editable", "crs", "crs definition", "crs description", "layer extent", "distance units", "layer type", "storage type", "geometry type", "feature count", "file path"]
+  "tags": ["property", "matching", "metadata", "layer name", "layer id", "title", "abstract", "keywords", "data url", "attribution", "attribution url", "layer source", "minimum scale", "min scale", "maximum scale", "max scale", "is editable", "crs", "crs definition", "crs description", "crs ellipsoid", "layer extent", "distance units", "layer type", "storage type", "geometry type", "feature count", "file path"]
```
Comment (DelazJ) [inline resources/function_help/json/layer_property:23]: Maybe do i misunderstand the concept of tags here, but I was wondering whether single words are not more suitable for tags? Results are returned as you type so we probably don't need long and composite expressions (with duplicate words)... Also, from a french translator perspective I see a number of preposition use that may return false positive.

```suggestion
  "tags": ["property", "matching", "metadata", "layer", "name", "id", "title", "abstract", "keywords", "data", "url", "attribution", "source", "minimum", "scale", "maximum", "editable", "crs", "definition", "description", "ellipsoid", "extent", "distance", "units", "type", "storage", "geometry", "feature", "count", "file", "path"]
```

Comment (gacarrillor) [reply resources/function_help/json/layer_property:23]: It's true that in general single-word tags should be preferred over compound ones, but compound tags may help describe better specific terms or properties. They help to be more precise and remove vagueness.

In this case, typing only words like `feature`, `id`, or even `layer` could be too broad as search. This is nowadays discouraged in the [documentation for plugin tags](https://docs.qgis.org/3.34/en/docs/pyqgis_developer_cookbook/plugins/releasing.html?highlight=choose%20tags%20with%20care).

Furthermore, a compound tag will be a match even if you search for one of its words, whereas some search widgets won't necessarily match combinations of single-word tags, which is the case of the Expression builder. BTW, that's why I'm proposing #55206 for plugins.

In summary, I think compound tags (for specific cases like well established layer properties, e.g., feature count) give us more benefits than drawbacks.

Comment (nyalldawson) [reply resources/function_help/json/layer_property:23]: What's the verdict here? Merge as is?

Comment (DelazJ) [reply resources/function_help/json/layer_property:23]: Oups, sorry guys!
> In this case, typing only words like feature, id, or even layer could be too broad as search. This is nowadays discouraged in the [documentation for plugin tags](https://docs.qgis.org/3.34/en/docs/pyqgis_developer_cookbook/plugins/releasing.html?highlight=choose%20tags%20with%20care).

I read that the docs advises not to use too generic words. It is not about single vs compound words, is it?
And while the words you mention here may look generic, they are of purpose, here IMHO.
And from my experience, I rarely type more than 4,5 letters in the expression builder before I find/browse the results in the list. At most a word. Never tried a compound expression but I understand it wouldn't have worked anyway. 
> BTW, that's why I'm proposing https://github.com/qgis/QGIS/pull/55206 for plugins.

Can it be extended to expression context?
> for specific cases like well established layer properties, e.g., feature count

Should it then be "feature_count" to make it clear and avoid confusion?

Well. I'm not versed in this topic, ignore the specific internals of the tool so i can't have strong opinion. I'm just raising questions and probably the way I'd proceed for their translation.
I'm OK with what @gacarrillor decides now. _for en_US_ 😄

Comment (gacarrillor) [reply resources/function_help/json/layer_property:23]: > I read that the docs advises not to use too generic words. It is not about single vs compound words, is it?

Exactly, and _too generic words_ might easily land if we always use single-word tags (e.g., `id`).

> Can it be extended to expression context?

Sure :+1:
Right now, I'm sort of understanding the commonalities between different QGIS search widgets.

> Should it then be "feature_count" to make it clear and avoid confusion?

That's a matter of taste. Anyways, in your example, the search component would have to split the tag and treat it as 2 words (i.e., normalize it). Otherwise, we might end up failing at finding results for function names/tags like `is_empty` when users type `is empty` (which is currently the case).

Additionally, since [spaces are allowed](https://github.com/agiudiceandrea/QGIS/blob/master/python/pyplugin_installer/installer_data.py#L71) in plugin tags, I think it'll be better not to change spaces by underscores in expression tags, for the sake of consistency.

> I'm just raising questions and probably the way I'd proceed for their translation.
I'm OK with what @gacarrillor decides now. for en_US 😄

Thanks for your questions, I find it valuable to discuss and address them, as we're doing now.
And, BTW, I'm also not the most appropriate person to decide this, specially for _en_US_ (!).

So, let's kindly invite @nyalldawson to join the discussion and/or to help us make a decision here.

**Comment by gacarrillor** [comment]
Just for reference, the comment was already addressed.

File: resources/function_help/json/layer_property (hunk 1 above)
Comment (nyalldawson) [reply resources/function_help/json/layer_property:23]: > Additionally, since [spaces are allowed](https://github.com/agiudiceandrea/QGIS/blob/master/python/pyplugin_installer/installer_data.py#L71) in plugin tags, I think it'll be better not to change spaces by underscores in expression tags, for the sake of consistency.

Agreed.

Comment (DelazJ) [reply resources/function_help/json/layer_property:23]: I've been testing #55378 which already introduced "space delimited tags" in expressions and I notice that we lose some functions whether we use _ or space, e.g.:
* crs_definition would return map_crs_definition and project_crs_definition functions but not layer_property, and the latter is the only one returned by "crs definition"
* geometry_type would return geometry_type function but not layer_property, and the latter is the only one returned by "geometry type"
* same thing for "file path", "distance units", "crs description", ... that all have their "underscore delimited version" as substring of an existing function (which sometime is a layer property)
* with this PR, "crs_ellipsoid" would return map_crs_ellipsoid and project_crs_ellipsoid functions but not layer_property, and the latter would be the only one returned by "crs ellipsoid"

Is it not a pity to not have both tags return the same results? Or at least have a tag that would return all corresponding functions? Or is that just the next step, and I'm in too much of a hurry 😄 ?

Comment (gacarrillor) [reply resources/function_help/json/layer_property:23]: Good observation and yes, that's part of further work to attempt to homogenize as much as we can (since sometimes it would not make sense, depending on the widget, context, etc.) the way QGIS search widgets work.

That is, without adding more tags to the functions you mentioned in your latest comment, a user will be able to find all matches, regardless of having underscores or spaces as separators.

**Review by lbartoletti — APPROVED** [review]

**Comment by lbartoletti** [comment]
@gacarrillor can you rebase your pr please?

**Comment by gacarrillor** [comment]
> @gacarrillor can you rebase your pr please?

Done @lbartoletti, and thanks for the review!
